src/astro/sql/operators/sql_dataframe_refactor.py

```diff
@@ -0,0 +1,113 @@
+from typing import Dict
+
+from airflow.decorators.base import DecoratedOperator
+
+from astro.databases import create_database
+from astro.settings import SCHEMA
+from astro.sql.table import Table as OldTable
+from astro.sql.table import TempTable
+from astro.sql.tables import Metadata
+from astro.sql.tables import Table as NewTable
+from astro.utils.dataframe_function_handler import DataframeFunctionHandler
+from astro.utils.table_handler_new import TableHandler
+
+
+class SqlDataframeOperator(DataframeFunctionHandler, DecoratedOperator, TableHandler):
+    def __init__(
+        self,
+        conn_id: str = "",
+        database: str = "",
+        schema: str = "",
+        warehouse: str = "",
+        role: str = "",
+        identifiers_as_lower: bool = True,
+        **kwargs,
+    ):
+        """
+        Converts a SQL table into a dataframe. Users can then give a python function that takes a dataframe as
+        one of its inputs and run that python function. Once that function has completed, the result is accessible
+        via the Taskflow API.
+
+        :param conn_id: Connection to the DB that you will pull the table from
+        :param database: Database for input table
+        :param schema:  schema for input table
+        :param warehouse: (Snowflake) Which warehouse to use for the input table
+        :param kwargs:
+        """
+        self.conn_id = conn_id
+        self.database = database
+        self.schema = schema
+        self.warehouse = warehouse
+        self.role = role
```

Comment:
Why do we need these properties to be associated to the operator?

Reply:
Yes, that's my thought, @dimberman ! IT would allow us to simplify our interfaces:
- `Table`
- `File`
- `pandas.Dataframe`